warp10/src/main/java/io/warp10/script/functions/TYPEOF.java

```diff
@@ -85,57 +85,69 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     return stack;
   }
 
-  public String typeof(Object o) {
+  public static String typeof(Object o) {
     if (null == o) {
       return (TYPE_NULL);
-    } else if (o instanceof String) {
+    } else if (o instanceof Typeofable) {
```

Comment:
Since the typeof() method of a typeofable is non-static, it needs an instance to operate.
For a Typeofable, typeof(Class c) tries to instanciate an object and can fail, in which case we return the canonical classname.